test-utils/to-match-string.ts

```diff
@@ -1,24 +1,40 @@
 import type { RegexSequence } from '../src/types';
 import { wrapRegExp } from './utils';
 
+interface MatchTypeOptions {
+  exactString: boolean;
+  substring?: string;
+}
+
 export function toMatchString(
   this: jest.MatcherContext,
   received: RegExp | RegexSequence,
   expected: string,
+  matchType?: MatchTypeOptions,
 ) {
   const receivedRegex = wrapRegExp(received);
   const matchResult = expected.match(receivedRegex);
+
+  let pass: boolean = false;
```

Comment:
I have trouble understanding what is the purpose of each of additional options. They seem not to be of any standard JS regex options for `String.match`, `RegExp.exec`, etc. Please explain their purpose as code comments.